sunpy/net/hek/utils.py

```diff
@@ -0,0 +1,197 @@
+    using units based on the specified coordinate frame.
+
+    Parameters
+    ----------
+    value: PolygonSkyRegion
+        A polygon defined using vertices in sky coordinates.
+    attribute: dict
+        An object from coord_properties.json
+    unit: str
+        The unit of the coordinates
+
+    Returns
+    -------
+    PolygonSkyRegion
+        A polygon defined using vertices in sky coordinates.
+
+    Raises
+    ------
+    IndexError
+        Because `value` does not contain the expected '((' and '))' substrings.
+    UnitConversionError
+        Because the units set by `coord1_unit` or `coord2_unit` are incompatible with the values being assigned.
+
+    """
+    coord1_unit = u.deg
+    coord2_unit = u.deg
+    if attribute["frame"] == "helioprojective":
+        coord1_unit = u.arcsec
+        coord2_unit = u.arcsec
+    elif attribute["frame"] == "heliocentric":
+        coord1_unit = u.R_sun # Nominal solar radius
+    elif attribute["frame"] == "icrs":
+        coord1_unit = get_unit(unit)
+        coord2_unit = get_unit(unit)
+
+    coordinates_str = value.split('((')[1].split('))')[0]
+    coord1_list = [float(coord.split()[0]) for coord in coordinates_str.split(',')] * coord1_unit
+    coord2_list = [float(coord.split()[1]) for coord in coordinates_str.split(',')] * coord2_unit
+    vertices = {}
+    if attribute["frame"] == "heliocentric":
+        vertices = SkyCoord(coord1_list, coord2_list, [1]* len(coord1_list) * u.AU, representation_type="cylindrical", frame="heliocentric")
+    else:
+        vertices = SkyCoord(coord1_list, coord2_list, frame=attribute["frame"])
+
+    return PolygonSkyRegion(vertices = vertices)
+
+# NOTE: Needs unit test
+def get_unit(unit):
+    """
+    Converts string into astropy unit.
+
+    Parameters
+    ----------
+    unit: str
+        The targeted unit
+
+    Returns
+    -------
+    unit
+        Astropy unit object (e.g. <class 'astropy.units.core.Unit'> or <class 'astropy.units.core.CompositeUnit'>)
+
+    Raises
+    ------
+    ValueError
+        Because `unit` did not parse as unit.
+
+    Notes
+    ----
+    For the complete list of HEK parameters: https://www.lmsal.com/hek/VOEvent_Spec.html
+
+    """
+    cm2 = u.def_unit("cm2", u.cm**3)
+    m2 = u.def_unit("m2", u.m**2)
+    m3 = u.def_unit("m3", u.m**3)
+
+    aliases = {
+        "steradian": u.sr,
+        "arcseconds": u.arcsec,
+        "degrees": u.deg,
+        "sec": u.s,
+        "emx": u.Mx,
+        "amperes": u.A,
+        "ergs": u.erg,
+        "cubic centimeter": u.ml,
+        "square centimeter": cm2,
+        "cubic meter": m3,
+        "square meter": m2,
+    }
+
+    with u.add_enabled_units([cm2, m2, m3]), u.set_enabled_aliases(aliases):
+        # If they are units of coordinates, it will have more than one unit,
+        # otherwise it will be just one unit.
+        # NOTE: There is an assumption that coord1_unit, coord2_unit and coord3_unit will be the same.
```

Comment:
```suggestion
        # Units for coordinate frames have more than one unit, otherwise it will be just one unit.
        # There is an assumption that coord1_unit, coord2_unit and coord3_unit are the same.
```